### ISSUE TITLE
"quasar build --skip-pkg" doesn't correctly update an iOS Cordova app

### ISSUE THREAD

**Issue by RohanTalip** [quasarframework/quasar#4686]
**Describe the bug**
Using the "--skip-pkg" (or "-s") argument to "quasar build" doesn't correctly update an iOS Cordova app.

(I was trying to use --skip-pkg because I wanted to speed up my build.)

**To Reproduce**
1. Build an app with `quasar build -m cordova -T ios`
2. Open the project in Xcode
3. From Xcode, run it on an iPhone device
4. Make some UI, logic or logging changes
5. Rebuild the app with `quasar build -m cordova -T ios --skip-pkg`
6. In Xcode, stop the app and run it again
7. Observe that the UI, logic or logging changes don't appear in the app.

**Expected behavior**
The UI, logic or logging changes made actually update the app.

**Platform (please complete the following information):**

```
$ quasar info

Operating System - Darwin(18.6.0) - darwin/x64
NodeJs - 10.15.3

Global packages
  NPM - 6.4.1
  yarn - 1.16.0
  @quasar/cli - 1.0.0
  cordova - 9.0.0 (cordova-lib@9.0.1)

Important local packages
  quasar - 1.0.4 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 1.0.4 -- Quasar Framework local CLI
  @quasar/extras - 1.1.4 -- Quasar Framework fonts, icons and animations
  vue - 2.6.10 -- Reactive, component-oriented view layer for modern web interfaces.
  vue-router - 3.0.7 -- Official router for Vue.js 2
  vuex - 3.1.1 -- state management for Vue.js
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.5.4 -- Babel compiler core.
  webpack - 4.28.4 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 3.7.2 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.6.2 -- Script for registering service worker, with hooks

Quasar App Extensions
  @quasar/quasar-app-extension-testing - 1.0.0 -- A Quasar App Extension for managing Test Harnesses
  @quasar/quasar-app-extension-testing-unit-jest - 1.0.0 -- A Quasar App Extension for running Jest tests

...
```

**Comment by rstoenescu** [comment]
Hi,

What --skip-pkg does is that it only builds the UI part (/src-cordova/www) folder and nothing else. I just checked and it seems to work correctly...

Does anyone else experience this?

**Comment by RohanTalip** [comment]
Hi Razvan (@rstoenescu),

I spent some time looking into this issue today, as I'm now seeing something similar (i.e. Cordova or Xcode caching weirdness) even without using `--skip-pkg` ... clearing the build folder in Xcode or deleting ~/Library/Developer/Xcode/DerivedData/ seems to fix this.

Anyway, I noticed today that there are actually 3 `www` directories under src-cordova:

1. src-cordova/www
2. src-cordova/platforms/ios/www
3. src-cordova/platforms/ios/build/emulator/My.app/www

When I specify `--skip-pkg`, the 2nd and 3rd directories are not updated and Xcode doesn't build/update my app with my recent changes to my Vue single file components, but when I don't specify `--skip-pkg` the 2nd and 3rd directories are updated.

So, I'm curious: is --skip-pkg supposed to do more for these other `www` directories?

If it isn't, what is the advantage of using `--skip-pkg` if it doesn't actually update the UI that gets used?  i.e. What does the `src-cordova/www` directory get used for, because it doesn't seem to be used in the built app if `--skip-pkg` is specified.

Has something changed in Cordova version 9 in terms of the directory layout?

**Comment by rstoenescu** [comment]
The 2nd and 3rd directories (and any other) should get updated by cordova CLI, based on the contents of /src-cordova/www. It's how cordova **should** work. Will investigate what goes wrong and what steps does Xcode skips. Seems to me like it's caching the /src-cordova/www folder and doesn't updates it...

**Comment by RohanTalip** [comment]
So,  if `--skip-pkg` is **not** specified, the [quasar-build script](https://github.com/quasarframework/quasar/blob/82a6b2fa0af5edf1d5fe37fb0696df8da4928e44/app/bin/quasar-build#L118) runs `cordova build` (with the target appended if `--target` or `-T` is also specified).  With `--skip-pkg` specified, quasar doesn't run the `cordova build` step.

According to the documentation for [cordova build](https://cordova.apache.org/docs/en/9.x/reference/cordova-cli/index.html#cordova-build-command), it is the equivalent to `cordova prepare` and `cordova compile`.

I tried running `quasar build -m cordova -T ios --skip-pkg` from my top-level application directory and `cordova prepare ios` from my src-cordova directory and then building my app from Xcode; it worked as I would expect.

The `cordova prepare ios` process copied the files from the `src-cordova/www` directory to the `src-cordova/platforms/ios/www` directory.

The "Copy www directory" build phase in Xcode runs `node cordova/lib/copy-www-build-step.js` to copy the `www` directory (and it seems to be the `src-cordova/platforms/ios/www` directory) to
~/Library/Developer/Xcode/DerivedData/MyApp-semirandomstring/Build/Products/Debug-iphonesimulator/MyApp.app/www .

The `src-cordova/platforms/ios/build/emulator/MyApp.app/www` was not updated.

I think quasar should be able to run `cordova prepare ${target}` (or similar) if `--skip-pkg` was specified.

Side note: At the moment, if I stop the simulator process from within Xcode and build the app again (from Xcode), I seem to get a weirdly cached older version of my app (that doesn't correspond to any of the 3 `www` directories mentioned above).  I have to clean the build folder in Xcode and build my app again in Xcode to see my recent changes.  As soon as I stop and rebuild my app in Xcode without making any other changes I get back to the weirdly cached older version of my app (even after clearing out the various directories mentioned in [this gist](https://gist.github.com/maciekish/66b6deaa7bc979d0a16c50784e16d697) and resetting the simulator via the Hardware > "Erase All Content and Settings..." menu option).  I don't know where this cached version is being kept.  If anyone else has seen this problem and/or knows how to fix it, I'd appreciate some tips.

**Comment by RohanTalip** [comment]
I think I've figured out the weird caching issue: I'm using Code Push and I wasn't incrementing the version number (just the build number) so I was getting an old version from Code Push once the app was stopped and restarted.

**Comment by rstoenescu** [comment]
Thank you for all the details!
But bottom line, is `cordova prepare` required?

**Comment by RohanTalip** [comment]
> is `cordova prepare` required?

Yes, it appears so.

**Comment by rstoenescu** [comment]
Thank you.
This will be available in quasar/app 1.0.6 (releasing ~tomorrow).

**Comment by RohanTalip** [comment]
Awesome, thanks Razvan.
